components/url_sanitizer/browser/url_sanitizer_service_unittest.cc

```diff
@@ -0,0 +1,143 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/url_sanitizer/browser/url_sanitizer_service.h"
+
+#include <vector>
+
+#include "base/containers/flat_set.h"
+#include "base/run_loop.h"
+#include "base/test/task_environment.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+namespace {
+const char kTestPatterns[] = R"(
+[
+  {
+    "include": [
+      "*://*.twitter.com/*"
+    ],
+    "params": [
+      "t"
+    ]
+  },
+  {
+    "include": [
+      "*://*/*"
+    ],
+    "exclude": [
+    ],
+    "params": [
+      "utm_content",
+      "utm_affiliate"
+    ]
+  },
+  {
+    "include": [
+      "https://dev-pages.bravesoftware.com/clean-urls/*"
+    ],
+    "exclude": [
+      "https://dev-pages.bravesoftware.com/clean-urls/exempted/*"
+    ],
+    "params": [
+      "brave_testing1",
+      "brave_testing2"
+    ]
+  }
+])";
+
+}  // namespace
+
+class URLSanitizerServiceUnitTest : public testing::Test,
+                                    public URLSanitizerService {
+ public:
+  URLSanitizerServiceUnitTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {}
+
+  void WaitInitialization(const std::string& json) {
+    base::RunLoop loop;
+    SetInitializationCallbackForTesting(loop.QuitClosure());
+    Initialize(json);
+    loop.Run();
+  }
+
+ private:
+  base::test::TaskEnvironment task_environment_;
+};
+
+TEST_F(URLSanitizerServiceUnitTest, StripQueryParameter) {
+  auto list = base::MakeFlatSet<std::string>(
+      std::vector<std::string>{"fbclid", "second"});
+
+  EXPECT_EQ(StripQueryParameter("fbclid=11&param1=1&second=2", list),
```

Comment:
I'd suggest a more complicated test to ensure that we don't break edge cases in the future:
- Before:`fbclid=11&fbclid2=ok&&param1=1&foo;bar=yes&second=2`
- After: `fbclid2=ok&&param1=1&foo;bar=yes`

Reply:
added it too